### ISSUE TITLE
Confused about code execution

### ISSUE THREAD

**Issue by peekxc** [quarto-dev/quarto#103]
For Quarto VSCode version 1.70 in the [changelog](https://github.com/quarto-dev/quarto/blob/main/apps/vscode/CHANGELOG.md#1700-release-on-26-february-2023) it was mentioned: 

> Cell execution and navigation for visual mode

For my visual editor I cannot execute cells though. I tried changing the keybinds but they seem to only work in the source mode. 

Am I misunderstanding the changelog comment or is this a bug on my end?

Versioning: 
> Versions: quarto v1.3.217 w/ VSCode extension v.1.71.0
> Platform: OS X 10.15.7

**Comment by jjallaire** [comment]
We haven't written up the docs on this (apologize for the delay!). When we do, we'll note that Python chunks currently can't be executed (I'm guessing that is what you are seeing). This is because we call the Jupyter extension to execute Python code and it in turn expects that an active text editor is in the foreground in order to work correctly. It should however currently work for R and Julia (we'd definitely like to make it work for Python but we need some outside help for that to happen)

**Comment by peekxc** [comment]
Ahh, I see, yes I was trying to execute Python code. Sounds good, I'll wait patiently; thanks for all your hard work regardless! 

Possibly related [this issue](https://github.com/microsoft/vscode-jupyter/issues/1237), which I think has also come up in the context of jupytext.

**Comment by jjallaire** [comment]
Yeah, that one is about their supporting other plain text formats for code execution. Here the visual editor is a "custom editor" so fails their check for `activeTextEditor` (which is pretty wound up in their overall approach, so not a one-liner to remedy).

**Comment by mcanouil** [comment]
Fore reference:

- https://github.com/quarto-dev/quarto-cli/discussions/8807

**Comment by mine-cetinkaya-rundel** [comment]
Currently, it is possible to execute Python cells individually in the visual editor in Positron, but not in VS Code.

**Comment by mcanouil** [comment]
FYI, positron repository: https://github.com/posit-dev/positron
